### PR TITLE
feat(ingestion/bigquery): Add ability to filter GCP project ingestion based on project labels

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_config.py [hunk 1]
```diff
@@ -214,6 +217,13 @@ def have_table_data_read_permission(self) -> bool:
             "Overrides `project_id_pattern`."
         ),
     )
+    project_labels: List[str] = Field(
+        default_factory=list,
+        description=(
+            "Ingests projects with the specified labels. Use this property if you want to specify what "
+            "projects to ingest based on project-level labels."
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_config.py:224]: please describe how this option interacts with project_id_pattern and other similar config options

**Review by hsheth2 — APPROVED** [review]
Mostly looks good, pending testing

**Comment by sid-acryl** [comment]
> Mostly looks good, pending testing

Testing is done and it is working fine with `project_labels`.

**Review by hsheth2 — APPROVED** [review]
Looks like there's a lint failure

**Pull request merged**